hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/AbstractYarnScheduler.java

```diff
@@ -722,11 +722,10 @@ protected void autoCorrectContainerAllocation(List<ResourceRequest> resourceRequ
       if (allocatedContainers != null) {
         for (RMContainer rmContainer : allocatedContainers) {
           if (extraContainers > 0) {
-            // Change the state of the container from ALLOCATED to EXPIRED since it is not required.
+            // Change the state of the container from ALLOCATED to RELEASED
+            // since it is not required.
             LOG.debug("Removing extra container:{}", rmContainer.getContainer());
-            completedContainer(rmContainer, SchedulerUtils.createAbnormalContainerStatus(
-                rmContainer.getContainerId(), SchedulerUtils.EXPIRED_CONTAINER),
-                RMContainerEventType.EXPIRE);
+            asyncContainerRelease(rmContainer);
```

Comment:
Will it cause the message at the end of the container to be different?

Reply:
Yes, the message will be updated from  "Container expired since it was unused" to "Container released by application", which I believe is more appropriate.